start defining what a stateful repository is (#1146)
jakartaee/data: spec/src/main/asciidoc/repository.asciidoc

A repository may have at most one resource accessor method.

A repository which declares a resource accessor method with return type `jakarta.persistence.EntityManager` is always a <<Stateful repositories,stateful repository>>.

> 

Author: Ugh. Oops. Well, we need to change that anyway, since `EntityAgent` is coming for certain.

A repository may have at most one resource accessor method.

> A repository which declares a resource accessor method with return type `jakarta.persistence.EntityManager` is always a <<Stateful repositories,stateful repository>>.

Reviewer: Here's an idea for turning this into a recommendation so that we maintain compatibility with Data 1.0,
```suggestion
A resource accessor method with return type `jakarta.persistence.EntityManager` should only be defined on a <<Stateful repositories,stateful repository>>.
```
Later, when we have entity agent, we could also write,
```
A resource accessor method with return type `jakarta.persistence.EntityAgent` must only be defined on a stateless repository.
```

[NOTE]
====
Implementations of Jakarta Data are encouraged to propagate a single persistence context within a given transaction across repositories which share a given datasource.
For example, a Jakarta Data provider backed by an implementation of Jakarta Persistence might take advantage of standard Jakarta EE transaction-scoped persistence context propagation.
====

==== Stateful repository lifecycle operations

A <<Lifecycle methods, lifecycle method>> of a stateful repository must be annotated with a lifecycle annotation specific to the stateful model.

The annotations `@Save`, `@Insert`, `@Update`, `@Delete` are used to define stateless repositories and must not be used to declare lifecycle methods of a stateful repository.
Instead, this specification defines the special lifecycle annotations `@Persist`, `@Merge`, `@Refresh`, `@Remove`, and `@Detach` for declaring lifecycle methods of stateful repositories.

A lifecycle method annotated `@Remove` or `@Refresh` only accepts managed entities associated with the persistence context underlying the repository.

>  On the other hand, a method annotated `@Persist`, `@Merge`, or `@Detach` also accepts unmanaged entities.

Reviewer: ```suggestion
On the other hand, a method annotated `@Persist`, `@Merge`, or `@Detach` also accepts unmanaged entities.
```
